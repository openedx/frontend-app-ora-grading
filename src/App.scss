// frontend-app-*/src/index.scss
@import "~@edx/brand/paragon/fonts";
@import "~@edx/brand/paragon/variables";
@import "~@openedx/paragon/scss/core/core";
@import "~@edx/brand/paragon/overrides";

$fa-font-path: "~font-awesome/fonts";
@import "~font-awesome/scss/font-awesome";

$input-focus-box-shadow: $input-box-shadow; // hack to get upgrade to paragon 4.0.0 to work

@import "~@edx/frontend-component-footer/dist/_footer";
@import "~@edx/frontend-component-header/dist/index";

#root {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  main {
    flex-grow: 1;
  }

  header {
    flex: 0 0 auto;

    .logo {
      display: block;
      box-sizing: content-box;
      position: relative;
      top: 0.1em;
      height: 1.75rem;
      margin-right: 1rem;
      img {
        display: block;
        height: 100%;
      }
    }
  }

  footer {
    flex: 0;
  }
}

#paragon-portal-root {
  .pgn__modal-layer {
    .pgn__modal-close-container {
      right: 1rem !important;
    }
  }
  .confirm-modal .pgn__modal-body {
    overflow: hidden;
  }
}
