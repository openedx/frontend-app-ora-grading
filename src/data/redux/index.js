import { combineReducers } from 'redux';

import { StrictDict } from '@src/utils';

import * as app from './app';
import * as grading from './grading';
import * as requests from './requests';
import * as submissions from './submissions';

export { default as thunkActions } from './thunkActions';

const modules = {
  app,
  grading,
  requests,
  submissions,
};

const moduleProps = (propName) => Object.keys(modules).reduce(
  (obj, moduleKey) => ({ ...obj, [moduleKey]: modules[moduleKey][propName] }),
  {},
);

const rootReducer = combineReducers(moduleProps('reducer'));

const actions = StrictDict(moduleProps('actions'));

const selectors = StrictDict(moduleProps('selectors'));

export { actions, selectors };

export default rootReducer;
