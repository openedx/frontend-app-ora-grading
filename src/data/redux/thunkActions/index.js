import { StrictDict } from '@src/utils';
// eslint-disable-next-line import/no-cycle
import app from './app';
import download from './download';
import grading from './grading';

export default StrictDict({
  app,
  download,
  grading,
});
