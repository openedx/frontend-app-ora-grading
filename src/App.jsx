import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import Footer from '@edx/frontend-component-footer';
import { LearningHeader as Header } from '@edx/frontend-component-header';

import { selectors } from 'data/redux';

import DemoWarning from 'containers/DemoWarning';
import CTA from 'containers/CTA';
import ListView from 'containers/ListView';
import { MathJaxContext } from 'better-react-mathjax';

import './App.scss';
import Head from './components/Head';

export const App = ({ courseMetadata, isEnabled }) => (
  <Router>
    <div>
      <Head />
      <Header
        courseTitle={courseMetadata.title}
        courseNumber={courseMetadata.number}
        courseOrg={courseMetadata.org}
      />
      {!isEnabled && <DemoWarning />}
      <CTA />
      <main>
        <MathJaxContext>
          <ListView />
        </MathJaxContext>
      </main>
      <Footer logo={process.env.LOGO_POWERED_BY_OPEN_EDX_URL_SVG} />
    </div>
  </Router>
);
App.defaultProps = {
  courseMetadata: {
    title: '',
    number: null,
    org: '',
  },
};
App.propTypes = {
  courseMetadata: PropTypes.shape({
    title: PropTypes.string,
    number: PropTypes.string,
    org: PropTypes.string,
  }),
  isEnabled: PropTypes.bool.isRequired,
};

export const mapStateToProps = (state) => ({
  courseMetadata: selectors.app.courseMetadata(state),
  isEnabled: selectors.app.isEnabled(state),
});

export default connect(mapStateToProps)(App);
