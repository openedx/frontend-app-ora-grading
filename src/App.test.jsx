import { render, screen } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { selectors } from 'data/redux';
import { App, mapStateToProps } from './App';

jest.unmock('react');
jest.unmock('@openedx/paragon');
jest.unmock('@edx/frontend-platform/i18n');

// we want to scope these tests to the App component, so we mock some child components to reduce complexity

jest.mock('@edx/frontend-platform/auth', () => ({
  getAuthenticatedHttpClient: jest.fn(),
  getLoginRedirectUrl: jest.fn(),
}));

jest.mock('@edx/frontend-component-footer', () => ({
  FooterSlot: () => <div data-testid="footer">Footer</div>,
}));

jest.mock('containers/ListView', () => function ListView() {
  return <div data-testid="list-view">List View</div>;
});

jest.mock('containers/DemoWarning', () => function DemoWarning() {
  return <div role="alert" data-testid="demo-warning">Demo Warning</div>;
});

jest.mock('data/redux', () => ({
  selectors: {
    app: {
      courseMetadata: jest.fn((state) => state.courseMetadata || {
        org: 'test-org',
        number: 'test-101',
        title: 'Test Course',
      }),
      isEnabled: jest.fn((state) => (state.isEnabled !== undefined ? state.isEnabled : true)),
    },
  },
}));

const renderWithIntl = (component) => render(
  <IntlProvider locale="en" messages={{}}>
    {component}
  </IntlProvider>,
);

describe('App component', () => {
  const defaultProps = {
    courseMetadata: {
      org: 'test-org',
      number: 'test-101',
      title: 'Test Course',
    },
    isEnabled: true,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders header with course metadata', () => {
    renderWithIntl(<App {...defaultProps} />);
    screen.debug();
    const org = screen.getByText((text) => text.includes('test-org'));
    expect(org).toBeInTheDocument();
    const title = screen.getByText('Test Course');
    expect(title).toBeInTheDocument();
  });

  it('renders main content', () => {
    renderWithIntl(<App {...defaultProps} />);

    const main = screen.getByTestId('main');
    expect(main).toBeInTheDocument();
  });

  it('does not render demo warning when enabled', () => {
    renderWithIntl(<App {...defaultProps} />);

    const demoWarning = screen.queryByRole('alert');
    expect(demoWarning).not.toBeInTheDocument();
  });

  it('renders demo warning when disabled', () => {
    renderWithIntl(<App {...defaultProps} isEnabled={false} />);

    const demoWarning = screen.getByRole('alert');
    expect(demoWarning).toBeInTheDocument();
  });

  describe('mapStateToProps', () => {
    it('maps state properties correctly', () => {
      const testState = { arbitraryState: 'some data' };
      const mapped = mapStateToProps(testState);

      expect(selectors.app.courseMetadata).toHaveBeenCalledWith(testState);
      expect(selectors.app.isEnabled).toHaveBeenCalledWith(testState);
      expect(mapped.courseMetadata).toEqual(selectors.app.courseMetadata(testState));
      expect(mapped.isEnabled).toEqual(selectors.app.isEnabled(testState));
    });
  });
});
