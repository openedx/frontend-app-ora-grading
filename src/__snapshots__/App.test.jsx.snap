// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App router component snapshot: disabled (show demo warning) 1`] = `
<MathJaxContext>
  <BrowserRouter>
    <div>
      <Head />
      <Header
        courseNumber="course-number"
        courseOrg="course-org"
        courseTitle="course-title"
        data-testid="header"
      />
      <DemoWarning />
      <CTA />
      <NotificationsBanner />
      <main
        data-testid="main"
      >
        <ListView />
      </main>
      <FooterSlot />
    </div>
  </BrowserRouter>
</MathJaxContext>
`;

exports[`App router component snapshot: enabled 1`] = `
<MathJaxContext>
  <BrowserRouter>
    <div>
      <Head />
      <Header
        courseNumber="course-number"
        courseOrg="course-org"
        courseTitle="course-title"
        data-testid="header"
      />
      <CTA />
      <NotificationsBanner />
      <main
        data-testid="main"
      >
        <ListView />
      </main>
      <FooterSlot />
    </div>
  </BrowserRouter>
</MathJaxContext>
`;
